app/controllers/posts_controller.rb

```diff
@@ -4,8 +4,28 @@ def index
     @posts = Post.where(author_id: @user.id)
   end
 
+  def all_posts
+    @user = params[:user_id]
+    @posts = Post.all
+  end
+
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @author_id = params[:author_id]
+  end
+
+  def create
+    param = params.require(:posts).permit(:title, :text)
+    @author = User.find(params[:author_id])
+    # @author = current_user
+    @post = Post.new(author: @author, title: param['title'], text: param['text'])
```

Comment:
- [ ] As stated in the requirements of this project, each post should be created on behalf of the currently logged-in user(current_user) of the app.